### ISSUE TITLE
Look at response headers to determine file name on "Send and Download"

### ISSUE THREAD

**Issue by adegroff** [a85/POSTMan-Chrome-Extension#2082]
1. Postman Version 4.2.1:
2. Chrome app
3. win / x86-32
4. Interceptor on and enabled 
5. Feature enhancement
6. Expected behaviour:

I have an endpoint that returns a file with the headers:

```
Content-Disposition →attachment; filename="filename.pdf"
Content-Type →application/pdf
```

Postman opens up a save dialog to save the response as "**response.pdf**"
It would be nice if Postman would default to save the file under the filename provided in the header (eg. "**filename.pdf**")

**Comment by a85** [comment]
This is fixed as part of the new Postman Runtime project. Details on our blog: http://blog.getpostman.com/2016/06/14/introducing-postman-runtime-more-power-better-control/

**Comment by anavioc** [comment]
I think doesn't work.
I had downloaded Postman Canary and name proposed for save file is "response.pdf"

Http Headers include:
Content-Disposition →attachment; filename="xxx"

**Comment by kmix** [comment]
I'm seeing the same thing in the Windows app. Current version is 4.8.1.

**Comment by sdnts** [comment]
@anavioc @kmix We're working on this, will let you know here when a fix is out :)

**Comment by anavioc** [comment]
@madebysid 👍 thanks!

**Comment by aku246** [comment]
Jaldi karo be... Yahan client se halat kharab h. Alok b gussa re h.

**Comment by koos42** [comment]
Just discovered this, it would be nice if Postman respected filename content-disposition header. I'm glad you all are working on this and look forward to the fruits of your labor. Thanks!

**Comment by sergiorykov** [comment]
It's still actual for **v4.9.2** :(

**Comment by albertomr86** [comment]
Same problem here!. Version 4.9.2.

**Comment by prem911** [comment]
I m on 4.10.2 and still get the same result response.pdf.pdf

**Comment by aku246** [comment]
try send and download.

**Comment by adegroff** [comment]
Update on issue:

running Postman for Chrome
Version 4.10.2
win / x86-32
Chrome 56.0.2924.87

performing a 'Send and Download'
`Cache-Control →no-cache
Content-Disposition →attachment; filename="another file name.pdf"
Content-Length →1391192
Content-Type →application/pdf`

Result is still saving the file as "_**response**_.pdf" and not "_**another file name**_.pdf"

**Comment by 05F3759DF** [comment]
Postman for Chrome
Version 5.0.0
OS X / x86-64
Chrome 58.0.3029.110

Same problem still exists!

**Comment by ericis** [comment]
*bump* just ran into this myself :)

**Comment by hvernanda** [comment]
Just want to "up" this issue

Access-Control-Allow-Headers →Origin, X-Requested-With, Content-Type, Accept
Access-Control-Allow-Origin →*
Connection →keep-alive
Content-Disposition →attachment;filename=yearly_data_user.csv
Content-Length →331
Content-Type →text/csv; charset=utf-8
Date →Thu, 20 Jul 2017 04:41:20 GMT
ETag →W/"14b-JN9w/1EOr+hx9LxGE1c2zaIwhuE"
X-Powered-By →Express

Result is still saving the file as "response.pdf" and not "another file yearly_data_user.csv"

**Comment by ArunKSY** [comment]
Request Headers:
cache-control:"no-cache"
Postman-Token:"9ae5d8d4-cd9a-49fe-9540-ba81d5da60c4"
Content-Type:"application/x-www-form-urlencoded"
User-Agent:"PostmanRuntime/6.2.5"
Accept:"*/*"
Host:"192.168.0.73:8093"
cookie:"JSESSIONID=131D2DE67E67791ADB2B42F285CA0CE7"
accept-encoding:"gzip, deflate"
content-length:60
Request Body:
Response Headers:
server:"Apache-Coyote/1.1"
content-disposition:"attachment; filename="120451295.png""
access-control-allow-origin:"*"
access-control-allow-credentials:"true"
access-control-allow-methods:"GET, POST, DELETE, PUT, OPTIONS, HEAD"
access-control-allow-headers:"Content-Type, Accept, X-Requested-With"
content-type:"image/*"
content-length:"2439"
date:"Thu, 17 Aug 2017 09:52:20 GMT"
Response Body:
�PNG


IHDR�.T"�Z	pHYsg��RtEXtSoftwareBarcode4J��v.	IDATx���W�K���s�9��	3fE|SQ�I�,("f�9��s�(
⋊�`���*j{{vW��r�=UPP���=�]������*srr�ǩʇ��Ɂ����rrr`�E��ׯ_1�{�.�x�b�߾}[ _�t)�o޼)�����ɶfϞ�6L��Çg�f�R9u�˗/�/7r�H���������*߸q���gϞe7o�L�1u��d�o��:S�L)�>n�8����#-?~��?|�P˫������Z>a��hҤI�~!ڕ��/&�W���?V�ɓ'A�u���g옌=:�O�6��|ԨQ%�sƌ�G��gΜY 3�B�^�*��+W�����U���,�*�%���.d�߿���������f��^�zYUUUŌ2:t(�O�>U��9|��ʖ�9�l�o߾�zÆ
�>}�ɍ5J�w�����6i�D�ܽ{7;u��ʧO�.j�:s�L���]�&�8w����ܹs��֭[�u�-ԦM-�z����m�V˭��k�.p)�СC�_�ڵk���+�QdމFd!���I���Cy�n�
ʛ5kV�9{��YP�y�z���5n�Xe�[�E&��cǎ���"Q�����Ȗ�o�<d��0��r`9�X,�ˁ��r`9�X�Xx�d+�/^d'N�2�<yRe���$�1���^�X޹sg�p��Pwٲeٚ5k��-9�cƌѶ�\M�<Y�իW�{�����ׯ�Ǐ���u��e,�e˖��� u6oެ޼A��6V�\��{����_�|�>|�����-4q��pMaÆP���:x������ŋ�={��������K��:�ǝ;w����g��>}���8���������Ǳcǆ�m۶�q�i�����ܯ_?��o�ڵA�#P��o�/��A�(��0:��"�˖hW�E>~��^�3gN���bA�u�+:��Ŋ�{�����ȖX�Y��Ξ=��v��%i�,���Rt�ԩ�z�;v�vؚ�ۈ-V޻���bInc�X�ْX)k���-[�[Ʀ��b\�6mZ�b�E+Ŝ���b!���Q�+��r`9�X,�ˁ��r`9�XuX(7�\)���cY<��)����䠨L����-�X(0�����o߮e�~�JN��!C
�����M��ħI[�(��cb�^w�7��Q����k_�K;đ��l޼y�
�����ݻwk���i�&�J .O��Gځ���
��7n�@�Ӷ�������ϟ�{�֭��x�~��C�㱁��c��������œ9t�Тy��q�٘��̳[I{�9󍌧Ŷ��U{+�at�9�
R��l�� �*�%ՠA��łx�R�P�>ܣG�P��Y��׮�XB-Z�HZ,V{V�T;b�,�},Ō��`@[,@`�}�T��a��`R����y�w-�����b�d���v+o�
F?�Q��+��r`9�X,�ˁ��r`9�X,�ˁUk��˳���kU��S�X13)���-Uݞ�A�Xy�2Kt;{J��1Y�},�n)�ǲ��,�9޻�SjO��XX�,�dO0Uݞ�<=�Hwt!���ˁ��r`9�X,�ˁ��r`9�X,�����bFAD��5���t*��©�\`K��T.�۠͹s��68�o�H -��w��U�I &
LEm���qx!_鋀M�"+Dp.i	(��㇖󮌉ᦂvm.�v���x�@�� Y��8��J������}��%۔�[��"Ʊ�{�޽� \���=�ǻ�l�>+
,�ٱc���2&6���Vrl��!��G�6�Vzl$u=/�y$R+�؈%��أ"B�]�>Vl��>��R'>6b��=6R�[�jU�bA�3e�9f�X�c��b�wl$/�L�b	x��i�y,�ˁ��r`9�X,�ˁ����6&��u8�-I:I��׌�"�#2�/�l�ΘI2	�Y��<%$��-����rXJ�N�_�vr<^�\r�]�t�x� n�Z�J�wZ��޽{�%K�h����ϟ?g555���K��R�b�
M̙J�	KRO�#�&c#�,�Y�d�ֵ'�d�臹�O�m��<�C��&�\�|�����z=a�X,a�L�U��>V�Ų{Z�]�
��b:�b�K1s*����ʳX��ڣ
!ږ��R�J&cSL�l���lb��X|�Y,˩��KIՍ���ˁ��r`9�X,�ˁ��r`�`�Ə{���7qm�E�d?�3�)�\�qo!�Sy���LƟ|�;f�ZBl�>�-�@�qo�R�٭�N}�;��.�q����Ƅ�O�y1���g�
�;�C��qo�ũ�l9�~�Z[�\��������'�����Ɂ��Tg�7�1�xGC.IEND�B`�
200

**Comment by nlaslett** [comment]
Bump.  I'm seeing this in Postman for Windows 5.2.  The filename is there in the Content-Disposition but Postman always wants to save the file as "download.pdf".

That's when I use the API that returns a download link.  When I use the API the returns the in-line content stream it wants to use "response.pdf", but that API call isn't returning Content-Disposition filename, so I can't blame Postman there.

(FYI, I'm using the Alfresco Document Management REST API to return files.)

**Comment by bensullivan** [comment]
Bump.

Content-Disposition →attachment; filename="advice-0542961c-c82d-408a-9934-1a020c098dc5.sheba"
Content-Type →application/xml
Date →Tue, 17 Oct 2017 06:35:31 GMT
Server →Jetty(9.4.4.v20170414)
Transfer-Encoding →chunked

Result is still saving the file as "response.xml" and not "advice-0542961c-c82d-408a-9934-1a020c098dc5.sheba"

I'm running version 5.3.0 on OSX 10.11.6

Any update on when this might be addressed?

Thanks

**Comment by sergioatzori** [comment]
Same here: Postman stand alone, 5.3.0.

Cache-Control →no-cache, private
Connection →Keep-Alive
Content-Disposition →attachment; filename="sistema-latruccheria.pdf"
Content-Encoding →gzip
Content-Type →application/octet-stream
Date →Tue, 31 Oct 2017 15:50:31 GMT
Keep-Alive →timeout=5, max=100
Server →Apache/2.4.10 (Debian)
Transfer-Encoding →chunked
Vary →Authorization,Accept-Encoding

When I "save & download", Postman saves response.bin

**Comment by emazv72** [comment]
Would be nice to have for my use case, like testing a REST download API.

Postman stand alone 5.3.2

Method: POST

Response headers:

Content-Disposition →attachment; filename="test-pubblica-2.pdf"
Content-Type →application/pdf
Date →Thu, 02 Nov 2017 17:22:31 GMT
Server →Apache-Coyote/1.1
Transfer-Encoding →chunked

Postman saves a generic "response.pdf"

**Comment by MattGurney** [comment]
I would also like this feature. It appears that Postman heeds the content-type header to determine the file name extension ("content-type:application/zip" produces response.zip) but ignores the content-disposition header filename property and makes the start of the file name always "response".

**Comment by m3talstorm** [comment]
Just hit this myself, would be great to see this added.

**Comment by gustavoponce7** [comment]
Any update on this implementation?

Currently using the latest version (**5.5.0**) and postman doesn't respect the filename defined in the header.

**Comment by esodot** [comment]
Postman **v5.5.0** is still **NOT** extracting the correct filename from Header with name "Content-Disposition". Hey guys, imho it's easy to fix this issue. Please give it a greater priority. Thank you. Kind regards esodot

**Comment by esodot** [comment]
please also for response of type "application/octet-stream", where current response is being saved to "response.bin" file. Thank you

**Comment by hoomb** [comment]
Still waiting for this feature. even after 1,5 Years

**Comment by kyleobrien91** [comment]
Any update on this? It's a disconcerting issue to have after 1.5 years. If you don't have the time, any pointers on where to start looking to perhaps submit a PR?

**Comment by grantyb** [comment]
I'd like to make the same offer as @kyleobrien91 . I'd love to see this fixed, and would be proud to contribute a fix if you could provide the starting guidance?

**Comment by ericweidl** [comment]
Bump.

**Comment by kdmashruwala** [comment]
Still issue with  Postman v6.0.10.

Content-Type →text/csv
Content-disposition →attachment; filename=export_2018-03-20_10-45-42.csv
Date →Tue, 20 Mar 2018 05:15:42 GMT
Transfer-Encoding →chunked
X-Application-Context →application:development

returns **response.csv** file instead of the **export_2018-03-20_10-45-42.csv** :(

**Comment by kristijan-ujevic-typeqast** [comment]
This needs to be fixed ASAP, people will lose time thinking it's a server problem.

**Comment by viniciusCamargo** [comment]
@kristijan-ujevic-typeqast I spent 2 hours last week debugging an API for this exact reason. :cry:

**Comment by drocx** [comment]
I also thought that it would have been a server problem...
Maybe someone could try to relieve the team of Postman and try to create a bug fix and submit it by pull request.

**Comment by inmate37** [comment]
Postman on Ubuntu 16.04. Still 

> response.xls

 when `Send and Download`

I thought it was my fault and not Postman's. Please fix this problem, it's really confusing !!!

**Comment by andrewpatrickburns** [comment]
Using Postman Version 6.0.10 on win32 10.0.16299 / x64

I'm setting the following headers:

> Content-Disposition →attachment; filename=2018-05-01-web-rolling.log
> Content-Length →4401110
> Content-Type →application/octet-stream

And it's returning **response.bin** in the save dialog.

**Comment by blkfhusky** [comment]
I got the same problem in mac app, version is 6.0.10

**Comment by prashantagarwal** [comment]
Hey Everyone. We are actively working on fixing this issue. I will keep you updated about its progress.

**Comment by makdeniss** [comment]
same here: response.bin instead of xlsx.

**Comment by sbanatech** [comment]
6.1.4 I still the issue, not saving the file in the exact output format, eg My response is a abc.lfg but saves as response.zip

**Comment by ramya-nadupalli** [comment]
6.1.4 I still see the issue, not saving the file in the exact output format, eg My response is abc.lfg but saves as response.zip

**Comment by dyre** [comment]
still an issue. glad to see it has been added to the roadmap... only 2 years after the first report

**Comment by johno1985** [comment]
Still an issue on version 6.2.3, Windows 10

**Comment by debodirno** [comment]
Facing the same issue on version 6.2.3, Mac.

**Comment by JoshuaVSherman** [comment]
facing same issue on Ubuntu 18.04.1 LTS, amazed that this issue is over two years old

**Comment by valentin-mentormate** [comment]
Same here. Postmen v6.2.5 :(

**Comment by siiberad** [comment]
Same here

**Comment by Alex99y** [comment]
Same!

**Comment by Prinsn** [comment]
another 2 hours clocked by me.

**Comment by vlaslaptev** [comment]
Same

**Comment by yanzishang** [comment]
Same!

**Comment by sujayvenaik** [comment]
Hey, 
We have addressed this issue and file name is now picked from the Content-Disposition header. These changes are up available on our canary channel (https://www.getpostman.com/canary) today and would soon be available on the stable channel. Do try these out in the canary and let us know.

**Comment by sujayvenaik** [comment]
Hey
This issue has been addressed and is available on the stable channel (v6.4.x).
You can update the app to the latest version. Do try it out and let us know.

Thanks again for writing in and reporting this issue.

**Comment by grantyb** [comment]
This works for me now (v6.4.2) 😃 . Thanks @sujayvenaik!

**Comment by bilalGit-kone** [comment]
> Hey
> This issue has been addressed and is available on the stable channel (v6.4.x).
> You can update the app to the latest version. Do try it out and let us know.
> 
> Thanks again for writing in and reporting this issue.

**Comment by riteshsingh2204** [comment]
using v6.7.4  and still getting it as Resopnse.pdf.

**Comment by gdrugan** [comment]
Using v7.36.1 and I get random name instead of name in content disposition at the first run.
But when I click Save Response to file I get the correct name.

**Comment by jayudhandha** [comment]
We are using latest version of postman v8.0.6 and we are getting filename as '0' while saving it.

Thanks!

**Comment by tinesoft** [comment]
Hi,

I'm using **v10.2.2** , and still the issue is present ( "Send and Download" saves file as "response.xxx", completely ignoring the header `"Content-Dispostion" --> "attachment; filename=<filename>"`) ....

**Comment by evilmao** [comment]
I'm using v10.5.2 is the latest version. , and still the issue is present 
![image](https://user-images.githubusercontent.com/22849885/206909196-4ec58dc8-e3f8-4b7b-a017-1a7bc56dc454.png)
![image](https://user-images.githubusercontent.com/22849885/206909217-19917741-79f1-4788-8859-32ef932a0c3d.png)

**Comment by ankitabylle** [comment]
@sujayvenaik 
on attaching the file it changes the filename to "/"test.pdf/""
from test.pdf.
I tried adding content disposition but its not working

**Comment by vldcreation** [comment]
I am using this kind of version:
Version 11.5.1
UI Version: 11.5.1-ui-240724-0915
Desktop Platform Version: 11.5.0 (11.5.0)

And the issue still present
<img width="1009" alt="image" src="https://github.com/user-attachments/assets/5a9a3dd1-759b-4f2f-96f9-bf02d23ad610">

**Comment by zils99** [comment]
i have same issue on ver 11.17.0.


![Screenshot 2024-10-15 232140](https://github.com/user-attachments/assets/d2922455-1d12-4ba1-bc01-813091972db4)

it name response.zip but Content-Disposition show Log_Aktivitas_3.zip

**Comment by DSchau** [comment]
@zils99 @ankitabylle can either of you share a simplified reproduction of your use case, or more detail? An ideal example would be a simplified project that shows the issue, or a way that the Postman team can reproduce the issue (so even just sharing full headers or a fuller sense of headers would be appreciated).

As an example, I created this simplified reproduction which _does_ work in Postman today, so I wonder what I did differently that the APIs you all are testing with do not show!

https://github.com/dschau/gh-2082

![Image](https://github.com/user-attachments/assets/06820588-402e-40dc-a5d3-93fb76014ed4)
